railties/test/generators/shared_generator_tests.rb

```diff
@@ -91,8 +104,10 @@ def test_template_is_executed_when_supplied_an_https_path
     template = %{ say "It works!" }
     template.instance_eval "def read; self; end" # Make the string respond to read
 
-    generator([destination_root], template: path).expects(:open).with(path, 'Accept' => 'application/x-thor-template').returns(template)
-    quietly { assert_match(/It works!/, capture(:stdout) { generator.invoke_all }) }
+    generator = generator([destination_root], template: path)
+    assert_called_with(generator, :open, [path, 'Accept' => 'application/x-thor-template'], returns: template) do
```

Comment:
Well, your reproduction is wrong. `foo` never yields anything. So `io` will be `nil` and then you get the error :grin:

I'll look into the real test now :smile: